src/dbnode/generated/proto/namespace/schema.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+package namespace;
+
+// SchemaMeta contains schema metadata.
+message SchemaMeta {
+    // message name as it appears in proto file descriptor.
+    string messageName = 1;
+}
+
+// SchemaOptions contains schema options for a namespace.
+message SchemaOptions {
+    // repo contains a history of deployed schema defintions in form of proto file descriptors.
+    FileDescriptorRepo repo = 1;
+    // schemas contains a set of named schemas for a namespace.
+    // a namespace can have multiple schemas.
+    // series identifies its schema by schema name (the map key) and a version (repo version).
+    map<string, SchemaMeta> schemas = 2;
+}
+
+// FileDescriptorRepo is a history of FileDescriptorSet sorted by version in descending order.
+message FileDescriptorRepo {
```

Comment:
Not a huge fan of the use of the word `Repo` here but I also don't have a better suggestion :P

Reply:
all right, I will use history.